core/lib/zksync_core/src/genesis.rs

```diff
@@ -101,24 +104,36 @@ pub async fn ensure_genesis_state(
     let genesis_root_hash = metadata.root_hash;
     let rollup_last_leaf_index = metadata.leaf_count + 1;
 
-    let block_commitment = L1BatchCommitment::new(
-        vec![],
+    let commitment_common_input = CommitmentCommonInput {
+        l2_to_l1_logs: Vec::new(),
         rollup_last_leaf_index,
-        genesis_root_hash,
-        vec![],
-        vec![],
-        base_system_contracts_hashes.bootloader,
-        base_system_contracts_hashes.default_aa,
-        vec![],
-        vec![],
-        H256::zero(),
-        H256::zero(),
-        *protocol_version,
-    );
+        rollup_root_hash: genesis_root_hash,
+        bootloader_code_hash: base_system_contracts_hashes.bootloader,
+        default_aa_code_hash: base_system_contracts_hashes.default_aa,
+        protocol_version: *protocol_version,
+    };
+    let commitment_input = if protocol_version.is_pre_boojum() {
```

Comment:
Nit: Maybe it makes sense to have constructor `CommitmentInput::for_genesis_batch(CommitmentCommonInput)` that would encapsulate this logic?